### ISSUE TITLE
Issue With Linking To The Library In VS2017

### ISSUE THREAD

**Issue by JohnMTrimbleIII** [aisouard/libwebrtc#57]
Project compiles just fine but I get the following unresolved errors when attempting to the run sample C++ file in a VS project. 

1>webrtc.lib(timeutils.obj) : error LNK2019: unresolved external symbol __imp_timeGetTime referenced in function "__int64 __cdecl rtc::SystemTimeNanos(void)" (?SystemTimeNanos@rtc@@YA_JXZ)
1>webrtc.lib(clock.obj) : error LNK2001: unresolved external symbol __imp_timeGetTime
1>webrtc.lib(trace_win.obj) : error LNK2001: unresolved external symbol __imp_timeGetTime
1>webrtc.lib(task_queue_win.obj) : error LNK2019: unresolved external symbol __imp_timeSetEvent referenced in function "public: bool __cdecl rtc::`anonymous namespace'::MultimediaTimer::StartOneShotTimer(unsigned int)" (?StartOneShotTimer@MultimediaTimer@?A0x0c2b7fe0@rtc@@QEAA_NI@Z)
1>webrtc.lib(task_queue_win.obj) : error LNK2019: unresolved external symbol __imp_timeKillEvent referenced in function "public: void __cdecl rtc::`anonymous namespace'::MultimediaTimer::Cancel(void)" (?Cancel@MultimediaTimer@?A0x0c2b7fe0@rtc@@QEAAXXZ)
1>webrtc.lib(task_queue_win.obj) : error LNK2019: unresolved external symbol __imp_timeBeginPeriod referenced in function "__int64 __cdecl rtc::`anonymous namespace'::GetTick(void)" (?GetTick@?A0x0c2b7fe0@rtc@@YA_JXZ)
1>webrtc.lib(clock.obj) : error LNK2001: unresolved external symbol __imp_timeBeginPeriod
1>webrtc.lib(task_queue_win.obj) : error LNK2019: unresolved external symbol __imp_timeEndPeriod referenced in function "__int64 __cdecl rtc::`anonymous namespace'::GetTick(void)" (?GetTick@?A0x0c2b7fe0@rtc@@YA_JXZ)
1>webrtc.lib(clock.obj) : error LNK2001: unresolved external symbol __imp_timeEndPeriod
1>webrtc.lib(httpcommon.obj) : error LNK2019: unresolved external symbol AcquireCredentialsHandleA referenced in function "enum rtc::HttpAuthResult __cdecl rtc::HttpAuthenticate(char const *,unsigned __int64,class rtc::SocketAddress const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class rtc::CryptString const &,struct rtc::HttpAuthContext * &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (?HttpAuthenticate@rtc@@YA?AW4HttpAuthResult@1@PEBD_KAEBVSocketAddress@1@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@33AEBVCryptString@1@AEAPEAUHttpAuthContext@1@AEAV45@6@Z)
1>webrtc.lib(httpcommon.obj) : error LNK2019: unresolved external symbol __imp_FreeCredentialsHandle referenced in function "public: virtual __cdecl rtc::NegotiateAuthContext::~NegotiateAuthContext(void)" (??1NegotiateAuthContext@rtc@@UEAA@XZ)
1>webrtc.lib(httpcommon.obj) : error LNK2019: unresolved external symbol InitializeSecurityContextA referenced in function "enum rtc::HttpAuthResult __cdecl rtc::HttpAuthenticate(char const *,unsigned __int64,class rtc::SocketAddress const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class rtc::CryptString const &,struct rtc::HttpAuthContext * &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (?HttpAuthenticate@rtc@@YA?AW4HttpAuthResult@1@PEBD_KAEBVSocketAddress@1@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@33AEBVCryptString@1@AEAPEAUHttpAuthContext@1@AEAV45@6@Z)
1>webrtc.lib(httpcommon.obj) : error LNK2019: unresolved external symbol CompleteAuthToken referenced in function "enum rtc::HttpAuthResult __cdecl rtc::HttpAuthenticate(char const *,unsigned __int64,class rtc::SocketAddress const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > const &,class rtc::CryptString const &,struct rtc::HttpAuthContext * &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &,class std::basic_string<char,struct std::char_traits<char>,class std::allocator<char> > &)" (?HttpAuthenticate@rtc@@YA?AW4HttpAuthResult@1@PEBD_KAEBVSocketAddress@1@AEBV?$basic_string@DU?$char_traits@D@std@@V?$allocator@D@2@@std@@33AEBVCryptString@1@AEAPEAUHttpAuthContext@1@AEAV45@6@Z)
1>webrtc.lib(httpcommon.obj) : error LNK2019: unresolved external symbol __imp_DeleteSecurityContext referenced in function "public: virtual __cdecl rtc::NegotiateAuthContext::~NegotiateAuthContext(void)" (??1NegotiateAuthContext@rtc@@UEAA@XZ)
1>webrtc.lib(audio_device_core_win.obj) : error LNK2019: unresolved external symbol MoInitMediaType referenced in function "private: int __cdecl webrtc::AudioDeviceWindowsCore::InitRecordingDMO(void)" (?InitRecordingDMO@AudioDeviceWindowsCore@webrtc@@AEAAHXZ)
1>webrtc.lib(audio_device_core_win.obj) : error LNK2019: unresolved external symbol MoFreeMediaType referenced in function "private: int __cdecl webrtc::AudioDeviceWindowsCore::InitRecordingDMO(void)" (?InitRecordingDMO@AudioDeviceWindowsCore@webrtc@@AEAAHXZ)
1>webrtc.lib(audio_device_core_win.obj) : error LNK2001: unresolved external symbol IID_IMediaBuffer
1>webrtc.lib(audio_device_core_win.obj) : error LNK2001: unresolved external symbol IID_IMediaObject
1>webrtc.lib(audio_device_core_win.obj) : error LNK2001: unresolved external symbol CLSID_CWMAudioAEC
1>C:\libwebrtc\sample\libwebrtcsample\x64\Debug\libwebrtcsample.exe : fatal error LNK1120: 15 unresolved externals

I have the following settings for preprocessor and I'm only including the webrtc.lib file. 
WIN32;
_DEBUG;_CONSOLE;
WEBRTC_WIN;
_CRT_SECURE_NO_DEPRECATE;
_CRT_NONSTDC_NO_DEPRECATE;
NOMINMAX;

Anyone have any ideas what's wrong?

**Comment by aisouard** [comment]
Hello,

Did you link the libraries listed in this file ? https://github.com/aisouard/libwebrtc/blob/dev/CMakeModules/FindLibraries.cmake#L28

**Comment by JohnMTrimbleIII** [comment]
Thanks for your help, if been trying to get VS to properly linked for days! So If anyone is having the same issues I'll include my dependency settings in VS. 

Additional Dependencies:
webrtc.lib;
msdmo.lib;
wmcodecdspuuid.lib;
dmoguids.lib;
crypt32.lib;
iphlpapi.lib;
ole32.lib;
secur32.lib;
winmm.lib;
ws2_32.lib;

Oh I also had to change my runtime library settings under C/C++ -> Code Generation to **Multi-threaded Debug (/MTd)**

I wasn't sure about how to use the **Using WebRTC in your project** section of the readme so that must be how I went wrong. 

Thanks again for your help!
